src/background/responders.ts

```diff
@@ -78,13 +78,35 @@ export const responders: { [T in Action['type']]: Responder<T> } = {
   CLICK_TAKE_SCREENSHOT(): Partial<AppState> {
     return {}
   },
+  CLICK_TWEET(state): Partial<AppState> {
+    if (state.tweeting?.state !== 'DONE') {
```

Comment:
Are these assertions for developers or for users? If it's for users, we should probably have some logic that prevents that. If it's for developers, I don't think it's necessary.

Reply:
They're for developers. It's definitely possible to be refactoring and mess this up. Also, by including the assertions in here typescript is able to know that `state.tweeting` has a certain data type in the subsequent lines